lib/matplotlib/legend.py

```diff
@@ -237,6 +237,10 @@ def _update_bbox_to_anchor(self, loc_in_canvas):
 title : str or None
     The legend's title. Default is no title (``None``).
 
+title_prop : None or `matplotlib.font_manager.FontProperties` or dict
```

Comment:
```suggestion
title_fontproperties : None or `matplotlib.font_manager.FontProperties` or dict
```
The naming is consistent with the parameter `prop`. Unfortunately, this is inconsistent with the rest of the library. In all other places we use `fontproperties`, not `prop`.

IMO `title_fontproperties` is to be preferred because it's clearer. It's a bit lengthy, but it's only used for fine-tuning and the passed value will likely be verbose as well. So we should favor clarity here.

Reply:
Yeah, I agree - good catch...

Reply:
Thanks for the suggestion. I have modified this parameter's name to title_fontproperties.